lib/src/constants.dart

```diff
@@ -43,6 +62,26 @@ abstract class ConstantReader {
   /// Throws [FormatException] if [isInt] is `false`.
   int get intValue;
 
+  /// Returns whether this constant represents a `List` literal.
```

Comment:
[nit] `get` methods should be documented like a variable or a field - with a noun so you can omit 'Returns'.

Similar below.